### DISCUSSION TITLE
Uploading an image as a blob to be used as a card image doesn't work !

### DISCUSSION THREAD

**Discussion by digfish** [MarshalX/atproto#212] [Questions] [answered]
Hi,

I want to automatically associate an image with a preview card inside a publication and so I tried:

`

    blob = bsky.com.atproto.repo.upload_blob(image_bytes)
    pprint(blob)
    blob.blob.mime_type = 'image/*'  

    embed_external = models.AppBskyEmbedExternal.Main(
        external=models.AppBskyEmbedExternal.External(
            title=preview_data.title,
            description=preview_data.description,
            uri=url,
            thumb=blob.blob,
        )
    )
`
with the following log:

```
2023-12-18 16:42:47,328 HTTP Request: POST https://bsky.social/xrpc/com.atproto.repo.uploadBlob "HTTP/1.1 200 OK"
Response(blob=BlobRef(mime_type='*/*', size=2483, ref=BlobRefLink(link='bafkreihg7iiatouv6hiqb7t5uharq3ikgcfd2ejbspw3bhxqp3led7xjea'), py_type='blob'))
2023-12-18 16:42:47,456 HTTP Request: POST https://bsky.social/xrpc/com.atproto.repo.createRecord "HTTP/1.1 400 Bad Request"
2023-12-18 16:42:47,457 Response(success=False, status_code=400, content=XrpcError(error='InvalidMimeType', message='Wrong type of file. It is */* but it must match image/*.'), headers=Headers({'date': 'Mon, 18 Dec 2023 16:42:46 GMT', 'content-type': 'application/json; charset=utf-8', 'content-length': '96', 'connection': 'keep-alive', 'x-powered-by': 'Express', 'access-control-allow-origin': '*', 'ratelimit-limit': '5000', 'ratelimit-remaining': '4961', 'ratelimit-reset': '1702921208', 'ratelimit-policy': '5000;w=3600', 'etag': 'W/"60-naTgCWHhjZ5FGkdXs552SxW2Nug"', 'vary': 'Accept-Encoding'}))
```
Apparently, the image is uploaded with the content-type of '*/*' but when associating the blob object returned from the `upload_blob` method with the `thumb` field it does not work, since it expects the blob of being of  content-type 'image/*' . I tried to force the mime_type of blob object to be of this value but it does not work either !

Thanks for your attention!

**Comment by MarshalX** [comment]
hi, are you sure that image_bytes is a valid image? also, i want to mention few things:
1. this is not SDK issue, your are receiving a backend error
2. typically backend works pretty well in guessing mime type 
3. ill convert this to discussion

pls show how you get content for image_bytes

**Reply by MarshalX** [reply, marked as answer]
Works for me. Post: https://bsky.app/profile/test.marshal.dev/post/3kgtmkuqnnj2s

Code:
```python
client = Client()
client.login(os.environ['USERNAME'], os.environ['PASSWORD'])

with open('cat2.jpg', 'rb') as f:
    cat_data = f.read()

upload = client.upload_blob(cat_data)
embed_external = models.AppBskyEmbedExternal.Main(
    external=models.AppBskyEmbedExternal.External(
        title='Test title',
        description='Test description',
        uri='https://atproto.blue',
        thumb=upload.blob,
    )
)

client.send_post('test external with thumb', embed=embed_external)
```

<details><summary>test image cat2.jpg</summary>
<p>

![cat2](https://github.com/MarshalX/atproto/assets/15520314/2348af71-7c3e-4efe-8c4c-cb1b891856be)

</p>
</details>

**Reply by digfish** [reply]
Hi, thanks for your code example. I was fetching an image from a webpage according to opengraph protocol returned in the [linkpreview](https://pypi.org/project/linkpreview/) library. It was returning a SVG image, so, since it is a text-based format, not a binary one, the blob was of generic  mime type (*/*).
So, I'll have to turn the SVG into a image binary format like JPEG or PNG.
Thanks for your code example, again.
Cheers

**Reply by neilstottler** [reply]
Holy moly the amount of digging I had to do to find this. This should really be on the docs page as an example.

**Reply by MarshalX** [reply]
@neilstottler like this? https://github.com/MarshalX/atproto/blob/main/examples/advanced_usage/send_ogp_link_card.py

**Reply by neilstottler** [reply]
Something like that. Honestly I'm the kind of person to want it directly on the docs page (not GH but that's good too) like this page https://atp.readthedocs.io/en/latest/atproto/atproto_client.models.app.bsky.embed.external.html with a link or something.
